user_modules/shared_testcases/testCasesForExternalUCS.lua

```diff
@@ -0,0 +1,179 @@
+local mobile_session = require("mobile_session")
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local json = require("modules/json")
+local sdl = require('SDL')
+
+local utils = { }
+
+-- [[ Variables ]]
+	utils.HMIAppIds = { }
+	utils.pts = nil
+
+-- [[ Functions ]]
+	function utils.createTableFromJsonFile(file)
+	  local f = io.open(file, "r")
+	  local content = f:read("*all")
+	  f:close()
+	  return json.decode(content)
+	end
+
+	function utils.createJsonFileFromTable(table, file)
+	  local f = io.open(file, "w")
+	  f:write(json.encode(table))
+	  f:close()
+	end
+
+	function utils.checkSDLStatus(test, expStatus)
+	  local actStatus = sdl:CheckStatusSDL()
+	  print("SDL status: " .. tostring(actStatus))
+	  if actStatus ~= expStatus then
+	    local msg = "Expected SDL status: " .. expStatus .. ", actual: " .. actStatus
+	    test:FailTestCase(msg)
+	  end
+	end
+
+	function utils.removeLPT()
+		local folderPath = commonPreconditions:GetPathToSDL() ..
+      commonFunctions:read_parameter_from_smart_device_link_ini("AppStorageFolder")
+    os.execute("rm -rf " .. folderPath)
+	end
+
+	function utils.removePTS()
+    local filePath = commonFunctions:read_parameter_from_smart_device_link_ini("SystemFilesPath") ..
+      "/" .. commonFunctions:read_parameter_from_smart_device_link_ini("PathToSnapshot")
+    os.execute("rm -rf " .. filePath)
+  end
+
+  local function updatePTU(appId)
+    utils.pts.policy_table.consumer_friendly_messages.messages = nil
+    utils.pts.policy_table.device_data = nil
+    utils.pts.policy_table.module_meta = nil
+    utils.pts.policy_table.usage_and_error_counts = nil
+    utils.pts.policy_table.app_policies[appId] = {
+      keep_context = false,
+      steal_focus = false,
+      priority = "NONE",
+      default_hmi = "NONE"
+    }
+    utils.pts.policy_table.app_policies[appId]["groups"] = { "Base-4", "Base-6" }
+    utils.pts.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+    utils.pts.policy_table.module_config.preloaded_pt = nil
+  end
+
+	local function ptu(test, id, status)
+    local policy_file_name = "PolicyTableUpdate"
+    local policy_file_path = commonFunctions:read_parameter_from_smart_device_link_ini("SystemFilesPath")
+    local ptu_file_name = os.tmpname()
+    local requestId = test.hmiConnection:SendRequest("SDL.GetURLS", { service = 7 })
+    EXPECT_HMIRESPONSE(requestId)
+    :Do(
+      function()
+        test.hmiConnection:SendNotification("BasicCommunication.OnSystemRequest",
+          { requestType = "PROPRIETARY", fileName = policy_file_name })
+        utils.createJsonFileFromTable(utils.pts, ptu_file_name)
+        EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", { status = "UPDATING" }, { status = status }):Times(2)
+        test["mobileSession"..id]:ExpectNotification("OnSystemRequest", { requestType = "PROPRIETARY" })
```

Comment:
@dboltovskyi, in case you don't specify to which application is sent BasicCommunication.OnSystemRequest, how you will be sure that OnSystemRequest will be sent for application("mobileSession"..id)